pravega/pravega-operator
# Issue 527: Support of multiple containers in pravega pods

### Change log description

We have a logic to check pod faulty when doing upgrade, but currently the logic only check the first container in the pod, which does not necessarily to be the pravega container, when customer inject sidecars.

### Purpose of the change

Fixes #527
Loop the containers and pick the correct container to check faulty.

### What the code does

Loop the containers and pick the correct container to check faulty.

### How to verify it

deploy Pravega and inject sidecars, and helm upgrade pravega, check if pravega can be upgraded successfully.
